diff --git a/net/bluetooth/hci_core.c b/net/bluetooth/hci_core.c
index 396696241..62a93e490 100644
--- a/net/bluetooth/hci_core.c
+++ b/net/bluetooth/hci_core.c
@@ -4679,7 +4679,7 @@ static void __check_timeout(struct hci_dev *hdev, unsigned int cnt, u8 type)
 
 	if (hci_dev_test_flag(hdev, HCI_UNCONFIGURED))
 		return;
-
+	/*
 	switch (type) {
 	case LE_LINK:
 		last_tx = hdev->le_last_tx;
@@ -4688,12 +4688,12 @@ static void __check_timeout(struct hci_dev *hdev, unsigned int cnt, u8 type)
 		last_tx = hdev->acl_last_tx;
 		break;
 	}
-
+	*/
 	/* tx timeout must be longer than maximum link supervision timeout
 	 * (40.9 seconds)
 	 */
-	if (!cnt && time_after(jiffies, last_tx + HCI_ACL_TX_TIMEOUT))
-		hci_link_tx_to(hdev, type);
+	if (!cnt && time_after(jiffies, hdev->acl_last_tx + HCI_ACL_TX_TIMEOUT))
+		hci_link_tx_to(hdev, ACL_LINK);
 }
 
 /* Schedule SCO */
