diff --git a/drivers/bluetooth/hci_ldisc.c b/drivers/bluetooth/hci_ldisc.c
index 2d960a5e3..05440ccc4 100644
--- a/drivers/bluetooth/hci_ldisc.c
+++ b/drivers/bluetooth/hci_ldisc.c
@@ -490,11 +490,6 @@ static int hci_uart_tty_open(struct tty_struct *tty)
 		BT_ERR("Can't allocate control structure");
 		return -ENFILE;
 	}
-	if (percpu_init_rwsem(&hu->proto_lock)) {
-		BT_ERR("Can't allocate semaphore structure");
-		kfree(hu);
-		return -ENOMEM;
-	}
 
 	tty->disc_data = hu;
 	hu->tty = tty;
@@ -506,7 +501,9 @@ static int hci_uart_tty_open(struct tty_struct *tty)
 
 	INIT_WORK(&hu->init_ready, hci_uart_init_work);
 	INIT_WORK(&hu->write_work, hci_uart_write_work);
-
+	
+	percpu_init_rwsem(&hu->proto_lock);
+	
 	/* Flush any pending characters in the driver */
 	tty_driver_flush_buffer(tty);
 
