diff --git a/drivers/bluetooth/hci_serdev.c b/drivers/bluetooth/hci_serdev.c
index 649d112ee..fd72d7809 100644
--- a/drivers/bluetooth/hci_serdev.c
+++ b/drivers/bluetooth/hci_serdev.c
@@ -300,14 +302,13 @@ int hci_uart_register_device(struct hci_uart *hu,
 	BT_DBG("");
 
 	serdev_device_set_client_ops(hu->serdev, &hci_serdev_client_ops);
-
-	if (percpu_init_rwsem(&hu->proto_lock))
-		return -ENOMEM;
-
+	
 	err = serdev_device_open(hu->serdev);
 	if (err)
-		goto err_rwsem;
-
+		return err;
+	
+	percpu_init_rwsem(&hu->proto_lock);
+	
 	err = p->open(hu);
 	if (err)
 		goto err_open;
@@ -379,8 +380,6 @@ int hci_uart_register_device(struct hci_uart *hu,
 	p->close(hu);
 err_open:
 	serdev_device_close(hu->serdev);
-err_rwsem:
-	percpu_free_rwsem(&hu->proto_lock);
 	return err;
 }
 EXPORT_SYMBOL_GPL(hci_uart_register_device);
@@ -402,6 +401,5 @@ void hci_uart_unregister_device(struct hci_uart *hu)
 		clear_bit(HCI_UART_PROTO_READY, &hu->flags);
 		serdev_device_close(hu->serdev);
 	}
-	percpu_free_rwsem(&hu->proto_lock);
 }
 EXPORT_SYMBOL_GPL(hci_uart_unregister_device);
